### PR TITLE
Several Updates to ESAPI

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
- Better Logging inside ESAPI
- Some tests
- xor-obfuscated parameter encryption
- bug in esapi special handling

### CODE REVIEW

File: src/tss2-esys/esys_crypto.c [hunk 1]
```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
...
+        /* XOR next data sub block with KDFa result  */
+        kdfa_byte_ptr = kdfa_result;
+        LOGBLOB_DEBUG(data, data_size, "XOR data");
+        for(size_t i = digest_size < rest_size ? digest_size : rest_size; i > 0; i--) {
+            LOG_DEBUG("II %zu", i);
+            *data++ ^= *kdfa_byte_ptr++;
+        }
+        LOGBLOB_DEBUG(data, data_size, "XOR data");
+        rest_size = rest_size < digest_size ? 0 : rest_size - digest_size;
+
```
Comment (tstruk) [inline src/tss2-esys/esys_crypto.c]: Extra line

File: src/tss2-esys/esys_crypto.c [hunk 2]
```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
...
+
+    }
+    return r;
+}
+
```
Comment (tstruk) [inline src/tss2-esys/esys_crypto.c]: Extra line

File: src/tss2-esys/esys_crypto.c [hunk 3]
```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
...
+{
+    TSS2_RC r;
+    uint32_t counter = 0;
+    BYTE  kdfa_result[TPM2_MAX_DIGEST_BUFFER];
+    size_t digest_size;
+    size_t data_size_bits = data_size * 8;
+    size_t rest_size = data_size;
+    BYTE *kdfa_byte_ptr;
+    r = iesys_crypto_hash_get_digest_size(hash_alg, &digest_size);
+    if (r != TSS2_RC_SUCCESS) {
```
Comment (tstruk) [inline src/tss2-esys/esys_crypto.c]: return_if_error(r, "Hash alg not supported");

File: src/tss2-esys/esys_crypto.c [hunk 4]
```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
...
+        LOG_ERROR("Hash alg not supported");
+        return r;
+    }
+    while(rest_size > 0) {
+        r = iesys_crypto_KDFa(hash_alg, key,
+                              key_size, "XOR",
+                              contextU, contextV,
+                              data_size_bits, &counter,
+                              kdfa_result, TRUE);
+        /* XOR next data sub block with KDFa result  */
```
Comment (tstruk) [inline src/tss2-esys/esys_crypto.c]: return_if_error(r, "iesys_crypto_KDFa failed");

File: src/tss2-esys/esys_crypto.c [hunk 5]
```diff
@@ -1127,3 +1128,47 @@ iesys_cryptogcry_sym_aes_decrypt(uint8_t * key,
...
+        kdfa_byte_ptr = kdfa_result;
+        LOGBLOB_DEBUG(data, data_size, "XOR data");
+        for(size_t i = digest_size < rest_size ? digest_size : rest_size; i > 0; i--) {
+            LOG_DEBUG("II %zu", i);
```
Comment (tstruk) [inline src/tss2-esys/esys_crypto.c]: Do we really want to print this counter here?

File: src/tss2-esys/esys_crypto.c (hunk 1 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_crypto.c]: Done

File: src/tss2-esys/esys_crypto.c (hunk 2 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_crypto.c]: Done

File: src/tss2-esys/esys_crypto.c (hunk 3 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_crypto.c]: done

File: src/tss2-esys/esys_crypto.c (hunk 4 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_crypto.c]: done

File: src/tss2-esys/esys_crypto.c (hunk 5 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_crypto.c]: Removed and changed surrounding LOGBLOB_DEBUGs to LOGBLOB_TRACEs

**Comment by AndreasFuchsSIT** [comment]
@tstruk updated...

**Pull request merged**